src/object_changeset.cpp

```diff
@@ -0,0 +1,156 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#include "object_changeset.hpp"
+
+using namespace realm;
+
+void ObjectChangeSet::insertions_add(ObjectKeyType obj)
+{
+    m_insertions.insert(obj);
+}
+
+void ObjectChangeSet::modifications_add(ObjectKeyType obj, ColKeyType col)
+{
+    // don't report modifications on new objects
+    if (m_insertions.find(obj) == m_insertions.end()) {
+        auto it_and_success = m_modifications.insert({obj, {col}});
```

Comment:
This is still just a slow and verbose way of writing `m_modifications[obj].insert(col);`.